pkg/parquetquery/iters.go

```diff
@@ -543,6 +543,10 @@ func (t *RowNumber) Next(repetitionLevel, definitionLevel int) {
 			t[7] = -1
 		}
 	}
+	if definitionLevel > MaxDefinitionLevel || definitionLevel < 0 {
+		level.Error(log.Logger).Log("msg", "definition level out of bound: should be [0:7] but got %d", definitionLevel)
+		panic("invalid definition level")
+	}
```

Comment:
For all cases where the definition level is valid this code will make two additional assertions on `definitionLevel` that have already proven to be false in the switch statements. Normally this would not bother me, but since `Next()` is very performance critical, it could be worth improving this. For example:
* Make sure that this code is never reached by adding a return after each inner switch statement
* Panic instead in a default case of every inner switch statement